[Doc][KubeRay] Add description tables for RayCluster Status in the observability doc (#47462)
ray-project/ray: doc/source/cluster/kubernetes/user-guides/observability.md

Use this command to redirect the operator's logs to a file called `operator-log`. Then search for errors in the file.

### Method 2: Check the status, and events of custom resources

```bash
kubectl describe [raycluster|rayjob|rayservice] $CUSTOM_RESOURCE_NAME -n $YOUR_NAMESPACE
```

After running this command, check events and the `state`, and `conditions` in the status of the custom resource for any errors and progresses.

> #### RayCluster Status State List

Reviewer: Status State is not clear for me.

```suggestion
#### RayCluster `.Status.State`
```

| State     | Description                                                                                                                            |
|-----------|----------------------------------------------------------------------------------------------------------------------------------------|

> | Ready     | The state will be set to Ready once all the Pods in the cluster are ready. The State will remain Ready until the cluster is suspended. |

Reviewer: ```suggestion
| Ready     | The state will be set to `Ready` once all the Pods in the cluster are ready. The `State` will remain `Ready` until the cluster is suspended. |
```

> | Suspended | The state will be set to Suspended when `Spec.Suspend = true` and all the Pods in the cluster are deleted.                             |

Reviewer: ```suggestion
| Suspended | The state will be set to `Suspended` when `Spec.Suspend` is set to true and all Pods in the cluster have been deleted.                             |
```

> #### RayCluster Status Condition List

Reviewer: ```suggestion
#### RayCluster `.Status.Conditions` 
```

> #### RayCluster `.Status.Conditions`

Reviewer: I remember that `.Status.Conditions` has a feature flag. Can you explain how to enable this feature?

Reviewer 2: Might help also add a warning header here that it's an alpha feature

Author: Both added.

<img width="1260" alt="image" src="https://github.com/user-attachments/assets/d9c9553e-60a5-4472-b623-f2ba4634c490">

|--------------------------|--------|--------------------------------|----------------------------------------------------------------------------------------------------------------------|

> | RayClusterProvisioned    | True   | AllPodRunningAndReadyFirstTime | Once all the Pods in the cluster are ready, this condition will be True and remain True even if some Pods die later. |

Reviewer: ```suggestion
| RayClusterProvisioned    | True   | AllPodRunningAndReadyFirstTime | Once all the Pods in the cluster are ready, this condition will be set to True and will remain True even if some Pods fail later. |
```

|                          | False  | RayClusterPodsProvisioning     |                                                                                                                      |
| RayClusterReplicaFailure | True   | FailedDeleteAllPods            | This condition will be set when there is a reconciliation error, otherwise the condition will be cleared.            |
|                          | True   | FailedDeleteHeadPod            | Please refer to the Reason and the Message of the condition for more detailed debugging information.                 |
|                          | True   | FailedCreateHeadPod            |                                                                                                                      |
|                          | True   | FailedDeleteWorkerPod          |                                                                                                                      |
|                          | True   | FailedCreateWorkerPod          |                                                                                                                      |

> | HeadPodReady             | True   | HeadPodRunningAndReady         | This condition will be True only the HeadPod is currently ready, otherwise it will be False.                         |

Reviewer: ```suggestion
| HeadPodReady             | True   | HeadPodRunningAndReady         | This condition will be True only if the HeadPod is currently ready; otherwise, it will be False.                         |
```

|                          | False  | HeadPodNotFound                |                                                                                                                      |

> |                          | False  | Unschedulable                  | Please refer to https://kubernetes.io/docs/concepts/scheduling-eviction/ for more details.                           |

Reviewer: How about removing the K8s built-in reasons (from Unschedulable to PreemptionByScheduler) from the documentation? Users can depend on the message to get more details about the reason. It is a bit hard for us to maintain reasons like `Unschedulable`, `SchedulingGated` ... etc.

Use this command to redirect the operator's logs to a file called `operator-log`. Then search for errors in the file.

### Method 2: Check the status and events of custom resources.

```bash
kubectl describe [raycluster|rayjob|rayservice] $CUSTOM_RESOURCE_NAME -n $YOUR_NAMESPACE
```

> After running this command, check events and the `state`, and `conditions` in the status of the custom resource for any errors and progresses.

Reviewer 3: ```suggestion
After running this command, check events and the `state`, and `conditions` in the status of the custom resource for any errors and progress.
```

#### RayCluster `.Status.State`

> The `.Status.State` field, which currently represents the cluster's situation, will be deprecated in the future due to its limited representation. Please use the new `Status.Conditions` as an alternative.

Reviewer 3: ```suggestion
The `.Status.State` field, which currently represents the cluster's situation, will be deprecated in the future due to its limited representation. Use the new `Status.Conditions` instead.
```

| State     | Description                                                                                                                            |
|-----------|----------------------------------------------------------------------------------------------------------------------------------------|

> | Ready     | The state will be set to `Ready` once all the Pods in the cluster are ready. The `State` will remain `Ready` until the cluster is suspended. |

Reviewer 3: ```suggestion
| Ready     | KubeRay sets the state to `Ready` once all the Pods in the cluster are ready. The `State` remains `Ready` until KubeRay suspends the cluster. |
```

> | Suspended | The state will be set to `Suspended` when `Spec.Suspend` is set to true and all Pods in the cluster have been deleted.                             |

Reviewer 3: ```suggestion
| Suspended | KubeRay sets the state to `Suspended` when it sets `Spec.Suspend` to true and deletes all Pods in the cluster.                             |
```

#### RayCluster `.Status.Conditions`

> Although `Status.State` can represent the cluster situation, it is still only a single field. By enabling the feature gate `RayClusterStatusConditions` on the KubeRay v1.2.1, you can access to new `Status.Conditions` for more detailed cluster history and states.

Reviewer 3: ```suggestion
Although `Status.State` can represent the cluster situation, it's still only a single field. By enabling the feature gate `RayClusterStatusConditions` on the KubeRay v1.2.1, you can access to new `Status.Conditions` for more detailed cluster history and states.
```

:::{warning}

> `RayClusterStatusConditions` is still an alpha feature and may undergo changes in the future.

Reviewer 3: ```suggestion
`RayClusterStatusConditions` is still an alpha feature and may change in the future.
```

:::

If you deployed KubeRay with Helm, then enable the `RayClusterStatusConditions` gate in the `featureGates` of your Helm values.

```bash
helm upgrade kuberay-operator kuberay/kuberay-operator --version 1.2.1 \
  --set featureGates\[0\].name=RayClusterStatusConditions \
  --set featureGates\[0\].enabled=true
```

> Or, just make your kuberay operator executable run with `--feature-gates=RayClusterStatusConditions=true` argument.

Reviewer 3: ```suggestion
Or, just make your KubeRay Operator executable run with `--feature-gates=RayClusterStatusConditions=true` argument.
```

| Type                     | Status | Reason                         | Description                                                                                                          |
|--------------------------|--------|--------------------------------|----------------------------------------------------------------------------------------------------------------------|

> | RayClusterProvisioned    | True   | AllPodRunningAndReadyFirstTime | Once all the Pods in the cluster are ready, this condition will be set to `True` and will remain `True` even if some Pods fail later. |

Reviewer 3: ```suggestion
| RayClusterProvisioned    | True   | AllPodRunningAndReadyFirstTime | Once all the Pods in the cluster are ready, this condition is set to `True` and remains `True` even if some Pods fail later. |
```

|                          | False  | RayClusterPodsProvisioning     |                                                                                                                      |

> | RayClusterReplicaFailure | True   | FailedDeleteAllPods            | This condition will be set to `True` when there is a reconciliation error, otherwise the condition will be cleared.            |

Reviewer 3: ```suggestion
| RayClusterReplicaFailure | True   | FailedDeleteAllPods            | KubeRay sets this condition to `True` when there's a reconciliation error, otherwise KubeRay clears the condition.            |
```

> |                          | True   | FailedDeleteHeadPod            | Please refer to the `Reason` and the `Message` of the condition for more detailed debugging information.                 |

Reviewer 3: ```suggestion
|                          | True   | FailedDeleteHeadPod            | See the `Reason` and the `Message` of the condition for more detailed debugging information.                 |
```

|                          | True   | FailedCreateHeadPod            |                                                                                                                      |
|                          | True   | FailedDeleteWorkerPod          |                                                                                                                      |
|                          | True   | FailedCreateWorkerPod          |                                                                                                                      |

> | HeadPodReady             | True   | HeadPodRunningAndReady         | This condition will be `True` only if the HeadPod is currently ready; otherwise, it will be `False`.                         |

Reviewer 3: ```suggestion
| HeadPodReady             | True   | HeadPodRunningAndReady         | This condition is `True` only if the HeadPod is currently ready; otherwise, it is `False`.                         |
```
